File: Patterns/TwoPointers/2Sum/2sum.cpp
    #include <iostream>
#include <unordered_map>
#include <vector>

std::vector<int> two_sum(const std::vector<int> &nums, int target)
{
    std::unordered_map<int, int> num_map; // Hash map to store {target - num: index}

    for (int i = 0; i < nums.size(); ++i)
    {
        if (num_map.find(nums[i]) != num_map.end())
        {                                 // Check if the current number exists in the map
            return {num_map[nums[i]], i}; // Return the indices
        }
        num_map[target - nums[i]] = i; // Store the difference and index
    }

    return {}; // No solution found
}

int main()
{
    std::vector<int> nums = {2, 7, 11, 15};
    int target = 9;

    std::vector<int> result = two_sum(nums, target);
    std::cout << "Indices of the two numbers: [" << result[0] << ", " << result[1] << "]\n";

    return 0;
}

File: Patterns/TwoPointers/2Sum/2sum.py
from typing import List,Dict
def two_sum(arr : List[int], target : int) -> List[int]:
    num_map : Dict[int,int] = {}
    
    for i,num in enumerate(arr):
        if num in num_map: # Look up keys
            return[num_map[num],i]
        num_map[target-num] = i
    
    return []

arr : List[int] = [2,7,11,15]
target : int = 9
print(f"Indicises of two numbers : {two_sum(arr,target)}")